src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Msmq/MessageQueue_Purge_Integration.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Msmq
+{
+    /// <summary>
+    /// Msmq calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+    AssemblyName = "System.Messaging",
+    TypeName = "System.Messaging.MessageQueue",
+    MethodName = "Purge",
+    ReturnTypeName = ClrNames.Void,
+    MinimumVersion = "4.0.0",
+    MaximumVersion = "4.*.*",
+    IntegrationName = MsmqConstants.IntegrationName)]
+    public class MessageQueue_Purge_Integration
+    {
+        private const string Command = MsmqConstants.MsmqPurgeCommand;
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TMessageQueue">Message queue</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TMessageQueue>(TMessageQueue instance)
+            where TMessageQueue : IMessageQueue
+        {
+            var scope = MsmqCommon.CreateScope(Tracer.Instance, Command, SpanKinds.Producer, instance);
```

Comment:
Since the purge action isn't directly producing or consuming a message, maybe we can use `SpanKinds.Client` instead of `SpanKinds.Producer` here